lib/pbench/server/api/resources/upload.py

```diff
@@ -105,26 +105,16 @@ def _stream(self, intake: Intake, request: Request) -> Access:
         Raises:
             APIAbort on failure
         """
-        try:
-            length_string = request.headers["Content-Length"]
-            content_length = int(length_string)
-        except KeyError as e:
-            # NOTE: Werkzeug is "smart" about header access, and knows that
-            # Content-Length is an integer. Therefore, a non-integer value
-            # will raise KeyError. It's virtually impossible to report the
-            # actual incorrect value as we'd just get a KeyError again.
-            raise APIAbort(
-                HTTPStatus.LENGTH_REQUIRED,
-                "Missing or invalid 'Content-Length' header",
-            ) from e
-        except ValueError as e:
-            # NOTE: Because of the way Werkzeug works, this should not be
-            # possible: if Content-Length isn't an integer, we'll see the
-            # KeyError. This however serves as a clarifying backup case.
-            raise APIAbort(
-                HTTPStatus.BAD_REQUEST,
-                f"Invalid 'Content-Length' header, not an integer ({length_string})",
-            ) from e
+        # Werkzeug returns an integer or None; either way, a false-y value is
+        # bad, so report it as an error.
+        content_length = request.content_length
+        if not content_length:
+            cl_val = request.headers.get("Content-Length")
+            msg = "Missing" if cl_val is None else "Invalid"
```

Comment:
I suspect this is pointless, as I believe Werkzeug will always return `None` here if the specified value wasn't an integer. I originally "documented" this with the `ValueError` case, but if you're determined to "streamline" this you might as well not preserve the pretense that we can tell the difference.

Reply:
Granted, this is from a while ago, but I believe that (at the time) I was convinced that `request.headers.get("Content-Length")` could return a value which was different from what `request.content_length` produced, and, therefore, that this code actually makes sense.  🙃

Reply:
Here's the code which my IDE thinks will be run for `request.content_length`:
```python
    @cached_property
    def content_length(self) -> t.Optional[int]:
        """The Content-Length entity-header field indicates the size of the
        entity-body in bytes or, in the case of the HEAD method, the size of
        the entity-body that would have been sent had the request been a
        GET.
        """
        if self.headers.get("Transfer-Encoding", "") == "chunked":
            return None

        content_length = self.headers.get("Content-Length")
        if content_length is not None:
            try:
                return max(0, int(content_length))
            except (ValueError, TypeError):
                pass

        return None
```
So, I think that referencing `request.headers.get("Content-Length")` will yield the value that the requestor provided, if any (but I could be wrong about that, given that Werkzeug also overrides `get()`...).

That said, I have no idea what `request.headers["content-length"]` produces in this scenario.  😆

Reply:
Yeah, but Werkzeug appears to be (and is documented to be) "smart" about the strings it allows into typed `Request` headers in the first place. As I said, I was unable to get it to store a non-integer value, regardless of the fact that it clearly *does* store and return the string value. I'm not sure why it has the `ValueError` here; maybe just in case someone does something silly like assigning it manually before calling `content_length`.

In any case, your logic is no sillier than the `ValueError` I had, even if your nonfunctional "documentation" is less obvious than mine, so "whatever". 😆

Reply:
OK, I've made another attempt, let me know what you think:  the change to the CUT is simpler; the impact is that uploading a zero-length file will now fail with `LENGTH_REQUIRED` instead of a `BAD_REQUEST` status, and the message indicates that zero is an invalid length without pointing out that it should be greater, because we're now catching it earlier in the path.